src/modalities/activation_checkpointing.py

```diff
@@ -20,10 +20,11 @@ def is_module_to_apply_activation_checkpointing(submodule: torch.nn.Module, acti
 
 
 def apply_activation_checkpointing_inplace(model: torch.nn.Module, activation_checkpointing_modules: List[str]):
-    activation_checkpointing_modules = [get_module_class_from_name(model, m) for m in activation_checkpointing_modules]
-    assert isinstance(model, FSDP), "activation checkpointing can only be applied to FSDP wrapped models!"
+    activation_checkpointing_module_types = [get_module_class_from_name(model, m) for m in activation_checkpointing_modules]
+    if not isinstance(model, FSDP)
+        raise ValueError("activation checkpointing can only be applied to FSDP wrapped models!")
     non_reentrant_wrapper = partial(checkpoint_wrapper, checkpoint_impl=CheckpointImpl.NO_REENTRANT, debug=False)
 
     apply_activation_checkpointing(
-        model, checkpoint_wrapper_fn=non_reentrant_wrapper, check_fn=is_module_to_apply_activation_checkpointing
+        model, checkpoint_wrapper_fn=non_reentrant_wrapper, check_fn=lambda submodule: is_module_to_apply_activation_checkpointing(submodule, activation_checkpointing_modules)
```

Comment:
```suggestion
        model, checkpoint_wrapper_fn=non_reentrant_wrapper, check_fn=lambda submodule: is_module_to_apply_activation_checkpointing(submodule, activation_checkpointing_modules_types)
```
shouldn't it be that way?